src/ServiceManagement/Compute/Commands.ServiceManagement/Common/DiagnosticsHelper.cs

```diff
@@ -551,7 +551,7 @@ private static bool TryGetStorageAccount(IStorageManagementClient storageClient,
             return storageAccount != null;
         }
 
-        private static AzureStorageContext CreateStorageContext(Uri blobEndpoint, Uri queueEndpoint, Uri tableEndpoint, Uri fileEndpoint,
```

Comment:
I don't think it's necessary to change this - AzureStorageContext implements IStorageContext - as long as this is not public, it should be fine to keep this internally.

Reply:
You're right. But it's just a simple change and I think it's better to use this interface wherever we can, unless a specific storage type is really needed.

Reply:
What I'm getting at is that there are specific methods for getting the storage credentials form an IStorageContext.  Are those used in this implementation?:

Reply:
I just looked at the code, no, the credentials is not actually being used.
I believe the reason we create the AzureStorageContext (which need the credential) is simply reusing the logic of EndPointSuffix property of AzureStorageContext, since we don't want to maintain this piece of logic.